File: fc-chat/demo/part3/ch4/message-system-add-push/src/main/java/net/prostars/messagesystem/entity/UserChannelId.java
package net.prostars.messagesystem.entity;

import java.io.Serializable;
import java.util.Objects;

public class UserChannelId implements Serializable {

  private Long userId;
  private Long channelId;

  public UserChannelId() {}

  public UserChannelId(Long userId, Long channelId) {
    this.userId = userId;
    this.channelId = channelId;
  }

  public Long getUserId() {
    return userId;
  }

  public Long getChannelId() {
    return channelId;
  }

  @Override
  public boolean equals(Object o) {
    if (o == null || getClass() != o.getClass()) return false;
    UserChannelId that = (UserChannelId) o;
    return Objects.equals(userId, that.userId) && Objects.equals(channelId, that.channelId);
  }

  @Override
  public int hashCode() {
    return Objects.hash(userId, channelId);
  }

  @Override
  public String toString() {
    return "ChannelId{UserId=%d, ChannelId=%d}".formatted(userId, channelId);
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/entity/UserChannelId.kt
package com.yscorp.withpush.messagesystem.entity

import java.io.Serializable
import java.util.*

class UserChannelId(
    val userId: Long,
    val channelId: Long
) : Serializable {

    override fun equals(o: Any?): Boolean {
        if (o == null || javaClass != o.javaClass) return false
        val that = o as UserChannelId
        return userId == that.userId && channelId == that.channelId
    }

    override fun hashCode(): Int {
        return Objects.hash(userId, channelId)
    }

    override fun toString(): String {
        return "ChannelId{UserId=%d, ChannelId=%d}".formatted(userId, channelId)
    }
}
